Bottom: bfb2d944e36c2b123cbd06ae10f11cf01c82dad8
Top:    b5cd069758de72e3c6716fde597031115fc6082e
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-03 14:49:07 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 10796fc..63d7ca4 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1187,7 +1187,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
